src/components/RightPane/CoursePane/SearchForm/TermSelector.tsx

```diff
@@ -32,6 +46,24 @@ class TermSelector extends PureComponent<TermSelectorProps> {
     handleChange = (event: ChangeEvent<{ name?: string | undefined; value: unknown }>) => {
         this.setState({ term: event.target.value });
         this.props.changeState(this.props.fieldName, event.target.value as string);
+    
+        let stateObj = { url: "url" };
+        const url = new URL(window.location.href)
+        const urlParam = new URLSearchParams(url.search);
+        urlParam.delete('term');
+        if (event.target.value as string != "" && event.target.value as string != null){
+            urlParam.append('term', event.target.value as string);
+            const new_url = `?${urlParam.toString()}`;
+            history.replaceState(stateObj, "url", "/" + new_url);
+        }else{
+            if (urlParam.toString() == "" || urlParam.toString() == null){
+                const new_url = `${urlParam.toString()}`;
+                history.replaceState(stateObj, "url", "/" + new_url);
+            }else{
+                const new_url = `?${urlParam.toString()}`;
+                history.replaceState(stateObj, "url", "/" + new_url);
```

Comment:
If it's possible, I'd recommend manipulating react-router-dom's history instead of the browser history. Maybe this API offers the functionality you're looking for? https://reactrouter.com/en/main/hooks/use-navigate

Reply:
So here I am trying to replace the browser's current url to a parameter appended status. I cannot find an API for react to do this smoothly. By changing the browser history, it can seamlessly change the url show on the web browser without reloading the whole website (Say kinda like redirect the website to the parameter appended url).